packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -1091,38 +1103,54 @@ def process_payload(self, payload: BaseTxPayload) -> None:
             )
 
         sender = payload.sender
-        if sender not in self.synchronized_data.participants:
+        if sender not in self.accepting_payloads_from:
             raise ABCIAppInternalError(
-                f"{sender} not in list of participants: {sorted(self.synchronized_data.participants)}"
+                f"{sender} not in list of participants: {sorted(self.accepting_payloads_from)}"
             )
 
         if sender in self.collection:
             raise ABCIAppInternalError(
                 f"sender {sender} has already sent value for round: {self.round_id}"
             )
 
+        if self._hash_length:
+            content = payload.data.get(self.payload_attribute)
+            if not content or len(content) % self._hash_length:
+                msg = f"Expecting serialized data of chunk size {self._hash_length}"
+                _logger.error(f"{msg}, got: {content} in {self.round_id}")
+                return
+
         self.collection[sender] = payload
 
     def check_payload(self, payload: BaseTxPayload) -> None:
         """Check Payload"""
+
+        # NOTE: the TransactionNotValidError is intercepted in ABCIRoundHandler.deliver_tx
+        #  which means it will be logged instead of raised
         if payload.round_count != self.synchronized_data.round_count:
             raise TransactionNotValidError(
                 f"Expected round count {self.synchronized_data.round_count} and got {payload.round_count}."
             )
 
-        sender_in_participant_set = (
-            payload.sender in self.synchronized_data.participants
-        )
+        sender_in_participant_set = payload.sender in self.accepting_payloads_from
         if not sender_in_participant_set:
             raise TransactionNotValidError(
-                f"{payload.sender} not in list of participants: {sorted(self.synchronized_data.participants)}"
+                f"{payload.sender} not in list of participants: {sorted(self.accepting_payloads_from)}"
             )
 
         if payload.sender in self.collection:
             raise TransactionNotValidError(
                 f"sender {payload.sender} has already sent value for round: {self.round_id}"
             )
 
+        if self._hash_length:
+            content = payload.data.get(self.payload_attribute)
+            if not content or len(content) % self._hash_length:
+                msg = f"Expecting serialized data of chunk size {self._hash_length}"
+                raise TransactionNotValidError(
+                    f"{msg}, got: {content} in {self.round_id}"
+                )
+
```

Comment:
here we raise `TransactionNotValidError`, which is intercepted in `ABCIRoundHandler.deliver_tx` here: 

https://github.com/valory-xyz/open-autonomy/blob/2822968f858def392ebfd4d48906ce1a81548234/packages/valory/skills/abstract_round_abci/handlers.py#L189-L208